### PR TITLE
P2: customize Join screen

### PR DESCRIPTION
**Pull request by annemirasol**
#### Changes proposed in this Pull Request

* Customize join/accept screens for P2 invites.

#### Testing instructions
Please change the host part of the invite link to `calypso.localhost:3000` when testing.

**NEW INTERFACE FOR P2**
* Open a valid P2 invite in incognito. You should see a customized "sign up and join" screen.
* Open a valid P2 invite while logged in. You should see a customized "join" screen.

**NO REGRESSSION**
* Invite functionality should not be affected, for both logged out and logged in use cases.
* Non-P2 invites should not be affected, and should still look and work the same.

Related to 1235-gh-Automattic/p2

#### Screenshots for translators

Screen Shot 2021-08-11 at 5 27 53 PM

Screen Shot 2021-08-11 at 5 28 33 PM

### CODE REVIEW

File: client/my-sites/invites/p2/invite-accept-logged-in.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+export function renderInviteAcceptForP2( props ) {
+	return (
+		<div>
+			<div className="invite-accept-logged-in__p2-logo">
+				<img src="/calypso/images/p2/logo.png" width="67" height="32" alt="P2 logo" />
+			</div>
+			<div className="invite-accept-logged-in__p2-learn-more">
+				<a href="https://p2help.wordpress.com">
+					{ props.translate( 'Learn more about P2', { context: 'link' } ) }
```
Comment (a8ci18n) [inline client/my-sites/invites/p2/invite-accept-logged-in.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Learn more about P2' )` *ES Score: 14*


:information_source: This string already exists without a **context**. Only add a context if the meaning of the string is very specific.

**Comment by annemirasol** [comment]
Screen Shot 2021-08-09 at 5 59 38 PM

File: client/my-sites/invites/p2/invite-accept-logged-out.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+			<div className="invite-accept-logged-out__form-container">
+				{ renderInviteAcceptFormHeader( { site: props.invite.site, translate: props.translate } ) }
+				<SignupForm
+					redirectToAfterLoginUrl={ window.location.href }
+					disabled={ props.isSubmitting }
+					formHeader=""
+					submitting={ props.isSubmitting }
+					save={ props.save }
+					submitForm={ props.submitForm }
+					submitButtonText={ props.isSubmitting ? '' : props.translate( 'Sign up and join' ) }
```
Comment (a8ci18n) [inline client/my-sites/invites/p2/invite-accept-logged-out.jsx:40]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'Sign Up & Join' )` *ES Score: 7*

File: client/my-sites/invites/p2/invite-accept-logged-out.jsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { renderInviteAcceptP2Logo } from './invite-accept-logo-line';
+import { renderInviteAcceptFormHeader } from './invite-accept-form-header';
+import { renderInviteAcceptFooter } from './invite-accept-footer';
+import LoggedOutFormLinks from 'calypso/components/logged-out-form/links';
+import LoggedOutFormLinkItem from 'calypso/components/logged-out-form/link-item';
+
+function renderFooterLink( props ) {
+	return (
+		<div className="invite-accept-logged-out__footer-link">
+			<div>{ props.translate( 'Already have a WordPress.com account? ' ) }</div>
```
Comment (a8ci18n) [inline client/my-sites/invites/p2/invite-accept-logged-out.jsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **45** times:
`translate( 'Already have a WordPress.com account?' )` *ES Score: 7*

**Comment by evilluendas** [comment]
Wonderful work Anne! I just pushed a few tweaks to the styling, and replaced the empty site icon with a P2 logo. After all, the P2 may have an icon but it's just unavailable to you so it's better not to use the default empty icon there (it's a similar case as with the xpost hovercards).

I've also removed the name of the P2 from the subtitle, since it's already mentioned in the title and it may become very repetitive if the name is long. 

Could you move the "Learn more about P2" from the top right and integrate in the description as a "Learn more" that opens in a new window, like this? I couldn't do it myself 😅 

image

Once that's done I'll tweak the overall top padding (specially on mobile) so I may try to replace the show/hide password icon via CSS too while I'm at it, so we use the updated icons.

Lastly, I noticed that when you're logged out (so you see the full form) the spinner doesn't appear in the button. Can you repro? It worked fine if you join while already logged in, though!

**Comment by annemirasol** [comment]
> Could you move the "Learn more about P2" from the top right and integrate in the description as a "Learn more" that opens in a new window, like this?

Just for the logged-out version?

UPDATE: Made "Learn more" inline for both versions.

**Comment by annemirasol** [comment]
Updated

✅ Moved "Learn more" link inline with the join explanation text.
✅ Fixed button spinner for signup screen


Screen Shot 2021-08-10 at 5 09 15 PM

**Comment by evilluendas** [comment]
I just pushed some small tweaks to the spacing on mobile, and swapped the password icons via CSS:

image

I think in terms of UI it's ready!

File: client/my-sites/invites/p2/invite-accept-logged-out.jsx (hunk 2 above)
Comment (annemirasol) [reply client/my-sites/invites/p2/invite-accept-logged-out.jsx:40]: Keeping this for consistency with P2 strings.

**Review by unDemian — APPROVED** [review]
Code looks great and tests as described. Should we also override the failures https://d.pr/i/xEufyr?

**Comment by annemirasol** [comment]
> Should we also override the failures https://d.pr/i/xEufyr?

cc @evilluendas On a follow-up PR, perhaps?

**Comment by lezama** [comment]
> On a follow-up PR, perhaps?

On a follow-up PR 👍

:shipit:

**Comment by annemirasol** [comment]
Test failure "Enter invalid search keyword" doesn't look related.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6393272

Thank you @annemirasol for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6393272).

**Pull request merged**